### PR TITLE
119907 cypress test 526ez confirmation page 

### LINKED ISSUE #119907: [Collab cycle requirement: E2E Test Participation] Create Cypress test for 526ez submission confirmation page
### User Need

As a developer I want to make sure that submission page and the copy of submission accordion is checked in the e2e cypress test so that we can catch bugs and be more confident about future changes. 

### Description

To the best of my knowledge current cypress tests don't cover the form submitted page and this needs to be addressed, including monitoring of the copy of submission accordion functionality 

For newer devs, please review Cypress Best Practices documentation here: https://depo-platform-documentation.scrollhelp.site/developer-docs/best-practices-for-using-cypress

Additionally, this may be a helpful resource: https://glebbahmutov.com/cypress-examples/commands/ (this is referencing Cypress@ 13.17.0 and our yarn.lock currently has Cypress @ 13.15.0 so there may be some slight differences).

### Tasks

- [ ] Review existing e2e tests (In this file: src/applications/disability-benefits/all-claims/tests/cypress.helpers.js)
- [ ] add new "chapter" under the pageHooks block for `'confirmation' `which is the URL path of the 526ez form submitted page (https://staging.va.gov/disability/file-disability-claim-form-21-526ez/confirmation)
- [ ] add check for confirmation flipper status: i.e. if the form data has the flipper on, then continue and check for these things:
  - [ ] assert that the confirmation page loads
  - [ ] assert that the accordion renders
  - [ ] assert that each piece of data is rendered and visible on the page

### Acceptance Criteria

- [ ]  Tests have been added 

### Definition of Done

- [ ] Meets acceptance criteria
- [ ] Passed all testing coverage
- [ ] Logging and monitoring in place
- [ ] Code or content reviewed (Internal)
- [ ] Reviewed and approved by product and/or design

### Refinement Checklist!

- [x] Add description, tasks, and acceptance criteria
- [ ] Add estimate
- [x] Add the proper labels, team, and priority
- [x] Add an Epic or Super Epic

### PR DESCRIPTION
**Pull request by Grendel78**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Added comprehensive Cypress E2E test for the 526EZ disability compensation claim confirmation page
- Test validates the confirmation review accordion feature controlled by the `disability_526_show_confirmation_review` feature toggle
- Created reusable Cypress commands to verify veteran details, conditions, toxic exposure, supporting evidence, and additional information displayed in the confirmation accordion
- Added test data fixture covering all form sections including rated/new disabilities, toxic exposure (Gulf War, Agent Orange, other hazards), medical facilities, and payment information
- Test includes `_13647Exception: true` to handle known platform limitation with nested `<dl>` elements when rendering complex array/object review data
- **Team**: DBC Core Form
- **Maintenance**: This test is maintained by DBC Core Form team as part of the 526EZ all-claims form
- **Feature Toggle**: `disability_526_show_confirmation_review` - no end date specified, toggle enables optional confirmation review accordion for enhanced user experience

## Related issue(s)

Closes department-of-veterans-affairs/va.gov-team#119907

### Old Behavior

Prior to this PR, there was no automated E2E test coverage for the 526EZ confirmation page with the confirmation review accordion feature

**New Cypress E2E Test** - Passing ✅

- Fills entire 526EZ form using comprehensive test data fixture
- Submits form and navigates to confirmation page
- Opens confirmation review accordion
- Validates 5 major sections with custom verification commands:
  - **Veteran Details**: Contact info, addresses, homeless status, service periods ✅
  - **Conditions**: 2 rated disabilities + 2 new disabilities with proper capitalization ✅
  - **Toxic Exposure**: Gulf War locations (Iraq, Kuwait), Agent Orange (Vietnam), other hazards (Asbestos, Radiation) with dates ✅
  - **Supporting Evidence**: VA facilities, private providers, uploaded documents ✅
  - **Additional Information**: Bank account, VA employee status, FDC program ✅

**Accessibility Testing** - Passing with documented exception ✅

- axe-core checks validate critical accessibility violations
- `_13647Exception: true` bypasses known platform limitation with nested `<dl>` elements

## What areas of the site does it impact?

- Cypress E2E testing for the confirmation page copy of submissions accordion

## Acceptance criteria

- [x] Added new 'confirmation' chapter under pageHooks block for the 526EZ form confirmation page URL path
- [x] Implemented check for `disability_526_show_confirmation_review` flipper status to conditionally run verification
- [x] Assert confirmation page loads successfully after form submission
- [x] Assert confirmation review accordion renders on the page
- [x] Assert all form data is rendered and visible within the accordion (veteran details, conditions, toxic exposure, supporting evidence, additional information)

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

- **Reusable Cypress Commands**: The PR adds 7 new custom Cypress commands for verifying confirmation page sections.
- **Test Data Completeness**: The test fixture covers the major scenarios (toxic exposure, multiple disabilities, VA/private facilities), but please flag any critical edge cases or data combinations that should be added for more comprehensive coverage.

### CODE REVIEW

**Review by kylesoskin — DISMISSED** [review]
That is a lot of tests. Nice!

**Review by powellkerry — DISMISSED** [review]

File: src/applications/disability-benefits/all-claims/tests/cypress.helpers.js [hunk 1]
```diff
@@ -882,4 +1222,52 @@ export const pageHooks = (cy, testOptions) => ({
...
+
+      // Wait for accordion content to be visible
+      cy.get('.confirmation-chapter-section-collection').should('be.visible');
+
+      cy.get('@testData').then(data => {
+        cy.verifyVeteranDetails(data);
+        cy.verifyConditions(data);
+        cy.verifyToxicExposure(data);
+        cy.verifySupportingEvidence(data);
+        cy.verifyAdditionalInformation(data);
```
Comment (tommasina-va) [inline src/applications/disability-benefits/all-claims/tests/cypress.helpers.js]: great thinking on pulling these out! I wonder if we can abstract out these pieces further for other tests too. (not actionable for this PR :))

File: src/applications/disability-benefits/all-claims/tests/fixtures/data/confirmation-test.json [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+      ],
+      "reservesNationalGuardService": {
+        "view:isTitle10Activated": false,
+        "obligationTermOfServiceDateRange": {
+          "from": "2005-01-01",
+          "to": "2010-01-01"
+        },
+        "unitName": "Reserve Unit 123"
+      },
+      "separationLocation": {
```
Comment (tommasina-va) [inline src/applications/disability-benefits/all-claims/tests/fixtures/data/confirmation-test.json]: This is a tricky piece of data. I forget how you generated this payload. Were you able to get this by selecting Reserves National Guard? We generally only see separation location for BDD but I see on line 7 says `"standardClaim": true,`which would mean this isn't a BDD claim. Just want to make sure it's reflective of the internal logic but I'm a little fuzzy on it. :)

**Review by tommasina-va — DISMISSED** [review]
Left a few questions but overall, impressive work! We have been needing someone to create a more tenable pattern for checking chunks of test data in Cypress helpers, and the `Cypress.Commands.add` logic seems like just what we need. 🏆

File: src/applications/disability-benefits/all-claims/tests/cypress.helpers.js [hunk 3]
```diff
@@ -882,4 +1222,52 @@ export const pageHooks = (cy, testOptions) => ({
...
   },
+
+  confirmation: ({ afterHook }) => {
+    cy.url().should('include', '/confirmation');
+    cy.get('va-button[text="Print this page for your records"]', {
+      timeout: 30000,
+    }).should('exist');
+
+    // Only verify detailed confirmation content if the confirmation review toggle is enabled
+    if (testOptions?.showConfirmationReview) {
```
[the remark this answers is not in the document]
Comment (Grendel78) [reply src/applications/disability-benefits/all-claims/tests/cypress.helpers.js]: Yay! Thank you for the suggestion.

File: src/applications/disability-benefits/all-claims/tests/cypress.helpers.js (hunk 1 above)
Comment (Grendel78) [reply src/applications/disability-benefits/all-claims/tests/cypress.helpers.js]: Yeah this kind of became a necessity. I was getting flagged for cognitive complexity so I had to break them out, but it also just made it make more sense to keep them organized and discreet.

**Review by tommasina-va — APPROVED** [review]
Looks great! 👏🏻

**Review by Cgfor3 — APPROVED** [review]
Thank you for thoughtfully going through this process. Setting us up for success!

**Comment by nihil2501** [comment]
@Grendel78 Is `src/applications/representative-form-upload/tests/e2e/form-upload.cypress.spec.js` failing? I see [this failure in CI/CD](https://github.com/department-of-veterans-affairs/vets-website/actions/runs/19377026432/job/55447108271) which also reproduces locally.
